libs/authorizenet/src/drivers/authorize-net.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, Inject } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+	import { 
+		DaffPaymentServiceInterface,
+		DaffPaymentTransformer,
+		DaffPaymentTransformerService
+	} from '@daffodil/checkout';
+
+import { DaffAuthorizeNetGenerateTokenSuccess, DaffAuthorizeNetGenerateTokenFailure } from '../actions/authorizenet.actions';
+import { AuthorizeNetResponse } from '../models/response/authorize-net-response';
+import { DaffAuthorizeNetTokenRequest } from '../models/request/authorize-net-token-request';
+
+declare var Accept: any;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffAuthorizeNetPaymentService implements DaffPaymentServiceInterface<DaffAuthorizeNetTokenRequest> {
+
+	constructor(
+		@Inject(DaffPaymentTransformer) public paymentTransformer: DaffPaymentTransformerService<DaffAuthorizeNetTokenRequest>,
+		private store: Store<any>
+	) {}
+
+	generateToken(paymentTokenRequest: DaffAuthorizeNetTokenRequest): string {
+		Accept.dispatchData(this.paymentTransformer.transformOut(paymentTokenRequest), this.responseHandler.bind(this));
+
+		return 'this string needs to be returned because authorize.net does not have a normal api';
+	}
+
+	private responseHandler(response: AuthorizeNetResponse) {
```

Comment:
This should be in an effect, not here, I understand why its is here, but hopefully the above refactor will allow it to be moved.

Reply:
done